azure-quantum/azure/quantum/qiskit/job.py

```diff
@@ -135,41 +137,61 @@ def _format_results(self):
                 job_result["data"] = self._format_unknown_results()
                 job_result["header"] = { "name": self._azure_job.details.name }
 
-        shots = self._azure_job.details.input_params[shots_key] if shots_key in self._azure_job.details.input_params else self._backend.options.get(shots_key)
+        shots = self._azure_job.details.input_params[shots_key] \
+            if shots_key in self._azure_job.details.input_params \
+            else self._backend.options.get(shots_key)
         job_result["shots"] = shots
         return job_result
 
-    def _format_ionq_results(self):
+    @staticmethod
+    def _draw_random_sample(sampler_seed, probabilities, shots):
+        rand = np.random.RandomState(sampler_seed)

```

Comment:
When you do this, normally this means that all the random numbers will return the same values, so calling the same method multiple times will return the same value (as such, not really drawing a random sample). Is this really the purpose?

Reply:
Yes, that is the purpose. If I specify a finite `sampler_seed` I want this function to always return the same values. This can be helpful for testing purposes.